rollup.base.config.js

```diff
@@ -24,26 +24,124 @@ function doCleanup() {
   })
 }
 
-const browserRollupConfigFactory = (banner, importCheckNames, targetType, theNameSpace, entryInputName, outputName, libVersion, isProduction, format = 'umd', postfix = '') => {
+const getNamespace = (prefix, namespaces, baseName, rootName) => {
+    var result = prefix + "var " + baseName + "=" + rootName;
+    if (namespaces.length > 0) {
+        for (let lp = 0; lp < namespaces.length; lp++) {
+            if (lp === 0) {
+                result += ", ";
+            } else {
+                result += ";\n" + prefix;
+            }
+            result += "nsKey=\"" + namespaces[lp] + "\", ";
+            result += baseName + "=" + baseName + "[nsKey]=(" + baseName + "[nsKey]||{})";
+        }
+    }
+
+    return result + ";\n";
+}
+
+const getCommonNamespace = (browserNs, gblNs) => {
+    var brTokens = browserNs.split(".");
+    var gblTokens = gblNs.split(".");
+    let common = [];
+    let idx = 0;
+    while (brTokens.length > idx && gblTokens.length > idx && brTokens[idx] === gblTokens[idx]) {
+        common.push(brTokens[idx]);
+        idx++;
+    }
+
+    return {
+        common: common,
+        browser: brTokens.slice(idx),
+        gbl: gblTokens.slice(idx)
+    };
+};
+
+const getIntro = (format, theNameSpace, moduleName, theVersion) => {
+    let theIntro = "";
+    if (format === "iife" || format === "umd") {
+        let nsTokens = getCommonNamespace(theNameSpace.browser, theNameSpace.gbl);
+        theIntro += "(function (global, factory) {\n";
+        let prefix = "    ";
+        theIntro += prefix + "var undef = \"undefined\";\n";
+        if (format === "umd") {
+            // UMD supports loading via requirejs and 
+            theIntro += prefix + "typeof exports === \"object\" && typeof module !== undef ? factory(exports) :\n";
```

Comment:
This is the main code that is injected to the top of every browser bundle (we don't run/change this for the main node (package.json) entrypoints -- currently as I don't believe we need this because of the way npm will wrap everything in a closure).

Once minified the new code is only around 200 bytes more, compared with "adding" the code and creating a new entrypoint (and adding code like this to every module) would add around 1k.

The previous UMD format generated by rollup for UMD format look like

```
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports) :
    typeof define === 'function' && define.amd ? define(['exports'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.Microsoft = global.Microsoft || {}, global.Microsoft.ApplicationInsights3 = global.Microsoft.ApplicationInsights3 || {})));
})(this, (function (exports) { 'use strict';
```
This new code changes this to look like

```
(function (global, factory) {
    var undef = "undefined";
    typeof exports === "object" && typeof module !== undef ? factory(exports) :
    typeof define === "function" && define.amd ? define(["exports"], factory) :
    (function(global){
        var nsKey, key, nm, theExports = {}, modName = "es5_ai_3_0_1", msMod="__ms$mod__";
        var mods={}, modDetail=mods[modName]={}, ver="3.0.1";
        var baseNs=global, nsKey="Microsoft", baseNs=baseNs[nsKey]=(baseNs[nsKey]||{});
        // Versioned namespace "Microsoft.ApplicationInsights3"
        var exportNs=baseNs, nsKey="ApplicationInsights3", exportNs=exportNs[nsKey]=(exportNs[nsKey]||{});
        // Global namespace "Microsoft.ApplicationInsights"
        var destNs=baseNs, nsKey="ApplicationInsights", destNs=destNs[nsKey]=(destNs[nsKey]||{});
        var expNsDetail=(exportNs[msMod]=(exportNs[msMod] || {})), expNameVer=(expNsDetail["v"]=(expNsDetail["v"] || []));
        var destNsDetail=(destNs[msMod]=(destNs[msMod] || {})), destNameVer=(destNsDetail["v"]=(destNsDetail["v"] || []));
        (destNsDetail["o"]=(destNsDetail["o"] || [])).push(mods);
        factory(theExports);
        for(var key in theExports) {
            // Always set the imported value into the "export" versioned namespace (last-write wins)
            nm="x", exportNs[key]=theExports[key], expNameVer[key]=ver;
            // Copy over any named element that is not already present (first-write wins)
            typeof destNs[key]===undef ? (nm="n", destNs[key]=theExports[key]) && (destNameVer[key]=ver) : !destNameVer[key] && (destNameVer[key]="---");
            (modDetail[nm] = (modDetail[nm] || [])).push(key);
        }
    })(typeof globalThis !== undef ? globalThis : global || self);
})(this, (function (exports) {
'use strict';
```

The "support" / "debug" additions added as part of this populate a `__ms$mod___` property to both namespaces, where the "global" one (Microsoft.ApplicationInsights in this case) will include a `v` object which enumerates every property name with the version that was applied to the namespace, for "unknown" (if an earlier version of the SDK was already loaded) this is set to `---`. It also contains an `o` which is an array of objects that includes the module name, `n` array with the property keys applied from this named module as well as an optional `x`  array with exported names from this module that where not applied to the global namespace

Reply:
![image](https://user-images.githubusercontent.com/54870357/236533999-147fd62e-4346-4473-8cf3-f0325014be39.png)